src/calling/haplotype_abundances.rs

```diff
@@ -85,6 +88,49 @@ impl Caller {
         })
     }
 
+    pub(crate) fn app_neg_binom(&self, seqname: String) -> Result<()> {
+        let hdf5 = &self.hdf5_reader;
+        let ids = hdf5
+            .dataset("aux/ids")?
+            .read_1d::<hdf5::types::FixedAscii<255>>()?;
+        let index = ids.iter().position(|x| x.as_str() == seqname).unwrap();
+        let num_bootstraps = hdf5.dataset("aux/num_bootstrap")?.read_1d::<i32>()?;
+        let seq_length = hdf5.dataset("aux/lengths")?.read_1d::<f64>()?;
+        let mut bootstraps = Vec::new();
+        for i in 0..num_bootstraps[0] {
+            let dataset = hdf5.dataset(&format!("bootstrap/bs{i}", i = i))?;
+            let est_counts = dataset.read_1d::<f64>()?;
+            let norm_counts = est_counts / &seq_length;
+            let norm_counts = norm_counts[index];
+            bootstraps.push(norm_counts);
+        }
+
+        //mean
+        let sum = bootstraps.iter().sum::<f64>();
+        let count = bootstraps.len();
+        let mean = sum / count as f64;
+
+        //std dev
+        let variance = bootstraps
+            .iter()
+            .map(|value| {
+                let diff = mean - (*value as f64);
+                diff * diff
+            })
+            .sum::<f64>()
+            / count as f64;
+        let std = variance.sqrt();
+
+        let theta = std / mean;
+
+        let mle_dataset = hdf5.dataset("est_counts")?.read_1d::<f64>()?;
+        let mle_norm = mle_dataset / &seq_length; //normalized mle counts by length
+        let mle = mle_norm[index];
+
+        let neg_bin = neg_binom(mle, mean, theta);
```

Comment:
The code looks correct. Now it is time to use/move this in the right context: we can start with implementing the actual prediction now.

1. the Caller struct should store mle, mean and theta for each haplotype. You can do this in a HashMap, which is initially empty and filled by calling a new method `estimate_parameters`. THis method should contain the code above in which you calculate mle, mean, and theta from the hdf5.
2. Then, we need to build a Bayesian model on top of this, which calculates the posterior probability of each combination of haplotype fractions. There is a framework for this in rust-bio: https://docs.rs/bio/0.37.1/bio/stats/bayesian/model/index.html. I suggest though that we get this particular part of the implementation started together. This should help you a bit to feel more comfortable with this kind of stuff. How about Sep 15, 9:30?
3. Once we have the posterior probability for each haplotype fraction, we can, for now, output each haplotype with its maximum a posteriori fraction.